libs/core/langchain_core/documents/base.py

```diff
@@ -1,47 +1,269 @@
 from __future__ import annotations
 
-from typing import Any, List, Literal, Optional
+import contextlib
+import mimetypes
+from io import BufferedReader, BytesIO
+from pathlib import PurePath
+from typing import Any, Generator, List, Literal, Mapping, Optional, Union, cast
 
 from langchain_core.load.serializable import Serializable
-from langchain_core.pydantic_v1 import Field
+from langchain_core.pydantic_v1 import Field, root_validator
 
+PathLike = Union[str, PurePath]
 
-class Document(Serializable):
-    """Class for storing a piece of text and associated metadata.
-
-    Example:
 
-        .. code-block:: python
+class BaseContent(Serializable):
```

Comment:
Not sure about this name since we use `page_content` already one level up - how about just `BaseMedia`?

Reply:
brilliant name!

Reply:
Do you think we need an optional name?